### ISSUE TITLE
Go 1.17 strict TLS APLN breaks TLS on TCP routers for non-HTTP protocols

### ISSUE THREAD

**Issue by andatche** [traefik/traefik#8376]
### Welcome!

- [x] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

The 2.5.0-rc6 release switched to go 1.17, which enforces strict TLS APLN checking by default. This breaks TLS on TCP routers in traefik where the upper-layer protocol is not HTTP because `config.NextProtos` of the TLS context only includes `h2`, `http/1.1` and the ACME TLS protocol.

https://github.com/traefik/traefik/blob/f15d05b22ffb752c5d9feeaed81346896cb2f29a/pkg/tls/tlsmanager.go#L236

I guess ideally `NextProtos` should be configurable on a per-router basis, or alternatively strict APLN checking should be disabled for TCP routers.

### What did you see instead?

TLS negotiation attempts fail with APLN error:

```
tls: client requested unsupported application protocols
```

### What version of Traefik are you using?

2.5.0

### What is your environment & configuration?

```yaml
tcp:
  routers:
    mosquitto_route:
      entryPoints:
      - "mqtt-mosquitto"
      service: mosquitto_service
      rule: "HostSNI(`*`)"
      tls:
        options: "mqtt@file"
        certResolver: le-2048
        domains:
          - main: "mqtt.<redacted>"
  services:
    mosquitto_service:
      loadBalancer:
        servers:
        - address: "mosquitto:1883"
```

### If applicable, please paste the log output in DEBUG level

time="2021-08-19T15:02:56Z" level=error msg="Error during connection: readfrom tcp 10.88.26.86:40794->10.88.26.76:1883: tls: client requested unsupported application protocols ([mqtt])"
time="2021-08-19T15:02:56Z" level=error msg="Error during connection: readfrom tcp 10.88.26.86:40796->10.88.26.76:1883: tls: client requested unsupported application protocols ([mqtt])"
time="2021-08-19T15:02:56Z" level=error msg="Error during connection: readfrom tcp 10.88.26.86:40798->10.88.26.76:1883: tls: client requested unsupported application protocols ([mqtt])"
time="2021-08-19T15:02:57Z" level=error msg="Error during connection: readfrom tcp 10.88.26.86:40800->10.88.26.76:1883: tls: client requested unsupported application protocols ([mqtt])"
time="2021-08-19T15:02:57Z" level=error msg="Error during connection: readfrom tcp 10.88.26.86:40802->10.88.26.76:1883: tls: client requested unsupported application protocols ([mqtt])"
time="2021-08-19T15:02:57Z" level=error msg="Error during connection: readfrom tcp 10.88.26.86:40804->10.88.26.76:1883: tls: client requested unsupported application protocols ([mqtt])"
time="2021-08-19T15:02:57Z" level=error msg="Error during connection: readfrom tcp 10.88.26.86:40806->10.88.26.76:1883: tls: client requested unsupported application protocols ([mqtt])"

**Comment by traefiker** [comment]
Closed by #8383.

**Comment by andatche** [comment]
Great, thanks for the quick fix!
